internal/pkg/service/buffer/definition/softdeletable.go

```diff
@@ -0,0 +1,15 @@
+package definition
+
+type SoftDeletableInterface interface {
+	IsDeleted() bool
+}
+
+type SoftDeletable struct {
+	Deleted bool `json:"deleted,omitempty"`
+	// DeletedWithParent is true when the object has not been deleted directly but was deleted together with its parent.
+	DeletedWithParent bool `json:"deletedWithParent,omitempty" validate:"excluded_if=Deleted false"`
+}
```

Comment:
Struktura, ktora bude `embedded` do `Branch,Source,Sink` entity, ktore maju podporovat soft-delete.

Reply:
`DeletedWithParent bool`

Ked sa zmaze parent object, napr. `branch`, ta sa musia zmazat aj vsetky pod objekty.
`branch -> source -> sink`.

Ked spravim `undelete branch`, tak sa ma vsetko obnovit.

No nemozem obnovit tie `sources/sinks`, ktore boli zmazane uz skor a zamerne (teda nie vramci mazania parent objektu).

Na rozlisenie tychto 2 pripadov sluzi `DeletedWithParent` flag.

-------------------

Skusil som pozriet ako to riesia a aky naming tomu davaju inde, [tu](https://laravel-news.com/laravel-soft-delete-parent) je to napr. `parent_deleted_at`.